### PR TITLE
Sushant/add support for jersey sdk

### PR DESCRIPTION
**Pull request by sushantdewan123**

### CODE REVIEW

File: src/main/java/com/wavefront/jersey/sdk/MetricNameUtils.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
+package com.wavefront.jersey.sdk;
```
Comment (srujann) [inline src/main/java/com/wavefront/jersey/sdk/MetricNameUtils.java]: package structure as com.wavefront.sdk.jersey similar to the wavefront-java-sdk?

File: src/main/java/com/wavefront/jersey/sdk/ApplicationTags.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Metadata about your application represented as tags in Wavefront.
+ *
+ * @author Sushant Dewan (sushant@wavefront.com).
+ */
+public class ApplicationTags {
```
Comment (srujann) [inline src/main/java/com/wavefront/jersey/sdk/ApplicationTags.java]: Can class this be in the core sdk as other sdk's will need the same tags?

File: .idea/workspace.xml [hunk 3]
```diff
@@ -0,0 +1,905 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (srujann) [inline .idea/workspace.xml]: this file might not be needed. similarly other files in .idea directory except for codeStyleSettings.xml and run configurations?

File: src/main/java/com/wavefront/jersey/sdk/ApplicationTags.java (hunk 2 above)
Comment (sushantdewan123) [reply src/main/java/com/wavefront/jersey/sdk/ApplicationTags.java]: Let's do that as a follow-up change. We can delete this class and fetch it from core-sdk.

File: src/main/java/com/wavefront/jersey/sdk/MetricNameUtils.java (hunk 1 above)
Comment (panghy) [reply src/main/java/com/wavefront/jersey/sdk/MetricNameUtils.java]: As long as we are consistent

File: src/main/java/com/wavefront/jersey/sdk/reporter/WavefrontJerseyReporter.java [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+  private WavefrontJerseyReporter(WavefrontInternalReporter wfReporter,
+                                  WavefrontMetricSender wavefrontMetricSender,
+                                  ApplicationTags applicationTags,
+                                  String source) {
+    this.wfReporter = wfReporter;
+    this.wavefrontMetricSender = wavefrontMetricSender;
+    this.applicationTags = applicationTags;
+    this.source = source;
+    ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1,
```
Comment (panghy) [inline src/main/java/com/wavefront/jersey/sdk/reporter/WavefrontJerseyReporter.java]: Can we extract this to a common class at this point, at some point we would pull this to T1.

File: src/main/java/com/wavefront/jersey/sdk/reporter/WavefrontJerseyReporter.java [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+      Logger.getLogger(WavefrontJerseyReporter.class.getCanonicalName());
+  private final WavefrontInternalReporter wfReporter;
+  private final WavefrontMetricSender wavefrontMetricSender;
+  private final ApplicationTags applicationTags;
+  private final String source;
+
+  private WavefrontJerseyReporter(WavefrontInternalReporter wfReporter,
+                                  WavefrontMetricSender wavefrontMetricSender,
+                                  ApplicationTags applicationTags,
+                                  String source) {
```
Comment (panghy) [inline src/main/java/com/wavefront/jersey/sdk/reporter/WavefrontJerseyReporter.java]: maybe check all of this to be nonnull?

File: src/main/java/com/wavefront/jersey/sdk/reporter/WavefrontJerseyReporter.java [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+  @Override
+  public void updateHistogram(MetricName metricName, long latencyMillis) {
+    wfReporter.newWavefrontHistogram(metricName).update(latencyMillis);
+  }
+
+  @Override
+  public void run() {
+    try {
+      wavefrontMetricSender.sendMetric(HEART_BEAT_METRIC, 1.0, System.currentTimeMillis(), source,
+          new HashMap<String, String>() {{
```
Comment (panghy) [inline src/main/java/com/wavefront/jersey/sdk/reporter/WavefrontJerseyReporter.java]: I feel like generating this map would only make sense if the tags are dynamic (and I doubt we would make it so), maybe just create an immutable map in the constructor?

File: src/test/java/com/wavefront/jersey/sdk/DropwizardApplicationReporterTest.java [hunk 7]
```diff
@@ -0,0 +1,477 @@
...
+import static com.wavefront.jersey.sdk.app.SampleApp.SERVICE;
+import static com.wavefront.jersey.sdk.app.SampleApp.SHARD;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Test class to test reported metric/histogram for Dropwizard (Jersey) apps requests/responses
+ *
+ * @author Sushant Dewan (sushant@wavefront.com).
+ */
+public class DropwizardApplicationReporterTest {
```
Comment (panghy) [inline src/test/java/com/wavefront/jersey/sdk/DropwizardApplicationReporterTest.java]: Let's not have Dropwizard mentioned in the Jersey SDK

File: src/main/java/com/wavefront/jersey/sdk/ApplicationTags.java [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+    private String shard;
+    private Map<String, String> customTags = new HashMap<>();
+
+    /**
+     * Builder to build ApplicationTags.
+     *
+     * @param application Name of the application.
+     * @param service     Name of the service.
+     */
+    public Builder(String application, String service) {
```
Comment (srujann) [inline src/main/java/com/wavefront/jersey/sdk/ApplicationTags.java]: may be check application and service are non null?

File: src/test/java/com/wavefront/jersey/sdk/DropwizardApplicationReporterTest.java [hunk 9]
```diff
@@ -0,0 +1,477 @@
...
+
+  private final SampleApp sampleApp = new SampleApp();
```
Comment (srujann) [inline src/test/java/com/wavefront/jersey/sdk/DropwizardApplicationReporterTest.java]: This class is missing in the project.

**Pull request merged**